Initial implemention of `{ <relationship>: { data: null } }` nullification

Samsinite opened the pull request:
Opening as a draft, will not merge, just looking for feedback, will have CleverCase use the latest commit on this branch after approval until it can be merged upstream.

This commit updates graphiti to be able to nullify relations only if
`null` is specified in the assocation payload. This will still need
feedback from the library maintainers, and after approval of design
tests will need written to test this feature.

Samsinite committed:
Initial implemention of `{ <relationship>: { data: null } }` nullification

This commit updates graphiti to be able to nullify relations only if
`null` is specified in the assocation payload. This will still need
feedback from the library maintainers, and after approval of design
tests will need written to test this feature.

jweakley reviewed (commented):
I didn't see anything that looks like it out of place. I think this would be good to propose to graphiti maintainers.

Samsinite committed:
...

Samsinite committed:
Update graphiti.gemspec for Ruby 3.1

Samsinite committed:
Update version.rb

Samsinite committed:
Update graphiti.gemspec

Samsinite committed:
Update version.rb

Samsinite committed:
Update serializer_relationships.rb

Samsinite committed:
Update graphiti.rb

Samsinite committed:
Update version.rb

Samsinite committed:
Samsinite patch ruby 3 2 (#2)

* Update to allow Ruby 3.2

* Update version.rb

Samsinite committed:
Update graphiti.gemspec

jweakley committed:
Bold Parameter in color method (#3)

Fixes issue where bold parameter was not being passed correctly to color method in ActiveSupport::LogSubscriber.

Samsinite committed:
Version bump to 1.2.36

jweakley committed:
Ruby 3.3 Compatibility (#4)

Ruby 3.3 Compatibility

Ensure the codebase is compatible with Ruby 3.3 by updating dependencies
and fixing deprecation warnings.

jweakley committed:
Update ActiveRecord adapter w/ support for Rails 7.2+ (#5)

In ActiveRecord 7.2 the `clear_active_connections` was (re)moved to the `connection_handler` property.
